lntest/itest/lnd_send_payment_multi_path.go

```diff
@@ -0,0 +1,271 @@
+// +build rpctest
+
+package itest
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+)
+
+// testSendPaymentMultiPath tests that we are able to successfully route a
+// payment using multiple shards across different paths.
+func testSendPaymentMultiPath(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	fmt.Printf("DEBUG %v: test start\n", time.Now())
+
+	// To ensure the payment goes through seperate paths, we'll set a
+	// channel size that can only carry one shard at a time. We'll divide
+	// the payment into 3 shards. We'll also set a small push amt, to make
+	// sure we don't violate channel reserves when we send a payment back.
+	const paymentAmt = btcutil.Amount(300000)
+
+	// Set up a network with three different paths Alice <-> Bob.
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	nodes := make(map[string]*lntest.HarnessNode)
+	nodes["alice"] = net.Alice
+	nodes["bob"] = net.Bob
+
+	// Create three more nodes.
+	nodeNames := []string{"carol", "dave", "eve"}
+	for _, name := range nodeNames {
+		node, err := net.NewNode(name, nil)
+		if err != nil {
+			t.Fatalf("unable to create new nodes: %v", err)
+		}
+		defer shutdownAndAssert(net, t, node)
+
+		nodes[name] = node
+	}
+
+	// Keep a list of all our active channels.
+	var networkChans []*lnrpc.ChannelPoint
+	var closeChannelFuncs []func()
+
+	// Open channels to/from the three new nodes.
+	openChannel := func(fromName, toName string, cap, push btcutil.Amount) {
+		fmt.Printf("Opening channel %v (%v) to %v (%v): cap=%v\n",
+			fromName, cap-push, toName, push, cap)
+
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		if err := net.ConnectNodes(ctxt, nodes[fromName], nodes[toName]); err != nil {
```

Comment:
in #3970 I had to connect the nodes before opening the channels to avoid the announcement propagation being flaky.

Reply:
Yes saw that. Probably going to extract a test context to share code with your test.

Reply:
Test context extracted